entity-service/src/main/kotlin/com/egm/stellio/entity/repository/Neo4jRepository.kt

```diff
@@ -255,48 +275,88 @@ class Neo4jRepository(
         return Pair(queryStatistics.nodesDeleted, queryStatistics.relationshipsDeleted)
     }
 
-    fun deleteEntityProperty(entityId: String, propertyName: String): Int {
+    fun deleteEntityProperty(subjectNodeInfo: SubjectNodeInfo, propertyName: String, datasetId: URI? = null, deleteAll: Boolean = false): Int {
         /**
          * Delete :
          *
          * 1. the property
          * 2. the properties of the property
          * 3. the relationships of the property
          */
-        val query = """
-            MATCH (entity:Entity { id: ${'$'}entityId })-[:HAS_VALUE]->(prop:Property { name: ${'$'}propertyName })
+
+        val query = if (deleteAll)

```

Comment:
your Cypher queries (in this function and in the next one) only differ by the initial match, you could avoid some duplication

Reply:
We create a deleteEntityAttribute method, we pass another argument to know if the attribute is a property or a relationship ?